samcli/commands/local/lib/local_lambda.py

```diff
@@ -152,8 +152,7 @@ def invoke(
             # trying to connect to the socket for Docker it would throw ContainerResponseException but now it's this.
             LOG.info(str(e))
         except OSError as os_error:
-            # pylint: disable=no-member
-            if hasattr(os_error, "winerror") and os_error.winerror == 1314:  # type: ignore
+            if getattr(os_error, "winerror", None) == 1314:
```

Comment:
Isn't there still a chance that this could be thrown in a non windows machine? Then would this change be compatible with Linux or Mac?

Reply:
No change of logic here, just simplified the check. If winerror isn't an attribute, getattr returns None, which is not 1314.